phpunit/script-dependencies-test.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @group script-dependencies
+ */
+class Test_Script_Dependencies extends WP_UnitTestCase {
+	public $bundled_scripts = [ 'wp-upload-media' ];
+
+	/**
+	 * Tests for accidental `wp-polyfill` script dependents.
+	 */
+	public function test_polyfill_dependents() {
+		$scripts            = wp_scripts();
+		$registered_scripts = $scripts->registered;
+		$dependents         = array();
+
+		// Iterate over all registered scripts, finding dependents of the `wp-polyfill` script.
+		// Based on private `WP_Scripts::get_dependents` method.
+		foreach ( $registered_scripts as $registered_handle => $args ) {
+			// Ignore bundled packages, they don't load separate polyfills.
+			if ( in_array( $registered_handle, $this->bundled_scripts, true ) ) {
+				continue;
+			}
```

Comment:
It is not ideal, but I couldn't find out why `wp-upload-media` is missing from `$dependents` on CI. However, we could skip checking for bundled packages as they are not enqueued separately.

The unit tests are passing without an issue locally. Happy to adjust further if anyone has ideas why this is only failing on CI.